lit_llama/lora.py

```diff
@@ -84,42 +85,34 @@ def reset_parameters(self):
             nn.init.zeros_(self.lora_B)
 
     def zero_pad(self, x):
+        x = x.transpose(0, 1)
         result = x.new_zeros((*x.shape[:-1], self.out_features))
         result = result.view(-1, self.out_features)
         result[:, self.lora_ind] = x.reshape(
             -1, self.out_features // len(self.enable_lora) * sum(self.enable_lora)
         )
-        return result.view((*x.shape[:-1], self.out_features))
+        return result.view((*x.shape[:-1], self.out_features)).transpose(0, 1)
```

Comment:
this bug was hidden behind train() and eval() never triggering the actual merge/unmerge. We would get shape mismatches without the transpose. 🤷